Add test statistics

matthewreyna opened the pull request:
These changes add the ability to use and define different test statistics with WExT.  The choice of test statistic is exposed as command-line argument for the `find_sets.py` script and defined in the `check_condition` function of  the `wext/saddlepoint.py` script.  Currently, three test statistics are available, but others are easy to define:

1. exclusivity (`-s exclusivity`);
2. any co-occurrence (`-s any-co-occurrence`); and
3. all co-occurrence (`-s all-co-occurrence`), which is equivalent to any co-occurrence for pairs and different for larger sets.

There is a simple example in the `examples/simple` directory that uses these test statistics, and it runs in a couple of seconds.  Also, the `find_sets.py` script only supports the WRE test with the enumeration search strategy, but the existing `find_exclusive_sets.py` script is unchanged.

braphael8 commented:
Can we fix the documentation to describe what all-occurrence and
pair-cooccurrence do, instead of saying how they are equivalent on pairs?
At the same time, we should document exclusivity.  Also, if -s is
unspecified, then is the default exclusivity?

Ben


On Tue, Oct 31, 2017 at 10:10 AM, matthewreyna <notifications@github.com>
wrote:

> Merged #20 <https://github.com/raphael-group/wext/pull/20>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/raphael-group/wext/pull/20#event-1318820563>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHGbDGUzmJsX8T_uteF3y6htUzWKurl9ks5sxyo5gaJpZM4QMx_D>
> .
>

matthewreyna commented:
I made several changes:
1. I updated the README, which links to a new page in the wiki that explicitly defines and describes these different statistics and how to define new ones.
2. I changed `pair-co-occurrence` to `any-co-occurrence`, which is more accurate (at least two mutated genes in a set instead of exactly two mutated genes) and better matches `all-co-occurrence`.
3. The `-s/--statistic` argument is required.
4. Users can still run `find_exclusive_sets.py`, which only uses exclusivity.  This script adds additional pre-processing steps, including only considering sets with more mutually exclusive mutations than co-occurring mutations, that we described in the paper but cannot easily generalize to arbitrary test statistics.
5. I had to revert the merge and reopen a pull request to add more changes to this branch and later merge it without conflicts.  Please add more comments to [the updated pull request](https://github.com/raphael-group/wext/pull/22).
